android/app/src/main/java/com/zulipmobile/notifications/FCMPushNotifications.kt

```diff
@@ -271,6 +264,12 @@ private fun updateNotification(
         .build()
     messagingStyle.addMessage(fcmMessage.content, fcmMessage.timeMs, sender)
 
+    // The linter sees that our tap action (at setContentIntent) is a service, not an
+    // activity.  The service will actually bring up an activity, so this is fine UI-wise,
+    // though understandably looks odd to the linter.
+    // TODO: Consider opening our activity directly; see:
+    //   https://developer.android.com/training/notify-user/navigation
+    @SuppressLint("LaunchActivityFromNotification")
```

Comment:
We just looked at this in the office, and found some more detail in a pretty obscure interaction in Android Studio. That contained the following:

> A Service or a BroadcastReceiver should not be used as an intermediate, because this can lead to significant performance problems, and as a result, this will not be allowed in Android 12.

So we probably want a `TODO(#5101)`.

Reply:
Yeah. From that text, it's not even clear if this only happens when targeting Android 12 -- read literally, it sounds like it's a behavior change that happens as soon as you're *running* on Android 12. There are always plenty of behavior changes like that too, but this particular change is one that I'd hope they would gate on the target version instead (because it's quite disruptive when unaccounted for, and because it's just performance and not something like a critical security or privacy hole).

Reply:
OK, indeed it's documented and is supposed to be conditioned on the target version:
https://developer.android.com/about/versions/12/behavior-changes-12#notification-trampolines